### PR TITLE
Add more metrics for mntr command

### PR DESCRIPTION
**Pull request by lzydmxy**
### Which issues of this PR fixes:

This PR is part of issue #168

### Change log:

Add those metrics 
- apply_read_request_time_ms: The time for request processor to process read request
- apply_write_request_time_ms: The time for request processor to process write request
- batch_size: Records the batch size of each batch accumulation for replication
- push_request_queue_time_ms: The time for push request from handler to dispatcher's request queue
- readlatency: Latency for read request. The timing start from when the server see the request until it leave final request processor
- updatelatency: Latency for write request. The timing start from when the server see the request until it leave final request processor

### CODE REVIEW

File: docs/how-to-monitor-and-manage.md [hunk 1]
```diff
@@ -82,6 +116,12 @@ zk_open_file_descriptor_count: current opening fd count
 zk_max_file_descriptor_count: max opening fd count
 zk_followers: follower count, only present on the leader
 zk_synced_followers: synced follower count, only present on the leader
+apply_read_request_time_ms: The time for request processor to process read request
```
Comment (JackyWoo) [inline docs/how-to-monitor-and-manage.md:119]: The time for request processor to process read request, log replication is not included for write request.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -66,7 +66,9 @@ if ((NOT USE_LIBCXX) AND COMPILER_CLANG AND OS_LINUX)
 endif()
 
 set(dbms_headers)
-set(dbms_sources)
+set(dbms_sources
```
Comment (JackyWoo) [inline src/CMakeLists.txt:69]: No need add metrics here.

File: src/Service/Metrics.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
+#include "Metrics.h"
```
Comment (JackyWoo) [inline src/Service/Metrics.cpp:1]: use <> instead of ""

File: docs/how-to-monitor-and-manage.md [hunk 4]
```diff
@@ -82,6 +116,12 @@ zk_open_file_descriptor_count: current opening fd count
...
+apply_write_request_time_ms: The time for request processor to process write request
+batch_size: Records the batch size of each batch accumulation for replication
```
Comment (JackyWoo) [inline docs/how-to-monitor-and-manage.md:121]: use log_replication_batch_size ?

File: tests/integration/test_four_word_command/test.py [hunk 5]
```diff
@@ -172,7 +172,6 @@ def test_cmd_mntr(started_cluster):
         assert int(result["zk_outstanding_requests"]) == 0
 
         assert result["zk_server_state"] == "leader"
-        assert result["zk_is_leader"] == "1"
```
Comment (JackyWoo) [inline tests/integration/test_four_word_command/test.py:175]: Please revert the change

File: src/Service/RequestProcessor.cpp [hunk 6]
```diff
@@ -246,6 +252,8 @@ void RequestProcessor::processCommittedRequest(size_t count)
                 /// apply request
                 applyRequest(committed_request);
                 committed_queue.pop();
+                auto current_time = getCurrentTimeMilliseconds();
+                Metrics::getMetrics().UPDATE_LATENCY->add(current_time - committed_request.create_time);
```
Comment (JackyWoo) [inline src/Service/RequestProcessor.cpp:256]: please update latency in `ConnectionHandler::updateStats`

File: src/Service/RequestProcessor.cpp [hunk 7]
```diff
@@ -425,6 +433,8 @@ void RequestProcessor::processReadRequests(RunnerId runner_id)
             if (session_request->request->isReadRequest())
             {
                 applyRequest(*session_request);
+                auto current_time = getCurrentTimeMilliseconds();
+                Metrics::getMetrics().READ_LATENCY->add(current_time - session_request->create_time);
```
Comment (JackyWoo) [inline src/Service/RequestProcessor.cpp:437]: Same with above

File: docs/how-to-monitor-and-manage.md [hunk 8]
```diff
@@ -36,28 +36,62 @@ Process uptime in seconds.
 `mntr` provide rich metrics and is the main monitoring command. The following is the output
 
 ```
-zk_version	RaftKeeper v2.0.4-e80e94979318adc94e81664053d782d0c9d7e60f, built on 2024-01-11 10:59:45 CST
+zk_version	RaftKeeper v2.0.5-96935adae174aaa430db0af14f823b0bf892f38e, built on 2024-04-15 17:19:43 CST
 zk_compatible_mode	zookeeper
 zk_avg_latency	1
-zk_max_latency	14598
+zk_max_latency	546
 zk_min_latency	0
```
Comment (JackyWoo) [inline docs/how-to-monitor-and-manage.md:43]: Please add a quantie of latency

File: src/Service/RequestProcessor.cpp (hunk 6 above)
Comment (lzydmxy) [reply src/Service/RequestProcessor.cpp:256]: If we need to update the statistics in `ConnectionHandler::updateStats`, we need to keep the `request` in the `response`. I will open another pr to fix it.

**Review by JackyWoo — APPROVED** [review]

**Comment by JackyWoo** [comment]
This PR will also close #206
